Mathlib/Analysis/Complex/Periodic.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Analysis.Complex.RemovableSingularity
+import Mathlib.Analysis.Calculus.InverseFunctionTheorem.Deriv
+import Mathlib.Order.Filter.ZeroAndBoundedAtFilter
+
+/-!
+# Periodic holomorphic functions
+
+We show that if `f : ℂ → ℂ` satisfies `f (z + h) = f z`, for some nonzero real `h`, then there is a
+function `F` such that `f z = F (exp (2 * π * I * z / h))` for all `z`; and if `f` is holomorphic
+at some `z`, then `F` is holomorphic at `exp (2 * π * I * z / h)`.
+
+We also show (using Riemann's removable singularity theorem) that if `f` is holomorphic and bounded
+for all sufficiently large `im z`, then `F` extends to a holomorphic function on a neighbourhood of
+`0`. As a consequence, if `f` tends to zero as `im z → ∞`, then in fact it decays *exponentially*
+to zero. These results are important in the theory of modular forms.
+-/
+
+open Complex Filter Asymptotics Function
+
+open scoped Real Topology
+
+noncomputable section
+
+local notation "I∞" => comap im atTop
+
+section q_param
+
+variable (h : ℝ)
+
+/-- Parameter for q-expansions, `q_param h z = exp (2 * π * I * z / h)` -/
+def q_param (z : ℂ) : ℂ := exp (2 * π * I * z / h)
```

Comment:
Should these not be camelCase?

Reply:
yes I was going to suggest `qParam`. I'm actually doing some of these changes now and will add them as a review.